### DISCUSSION TITLE
translation optimization

### DISCUSSION THREAD

**Discussion by Musho-T** [vinissimus/next-translate#1000] [General]
it is possible that only those words that we use in the page got into ssr, and not the whole json ?

**Comment by aralroca** [comment]
Not directly. Actually, the way to do it is to use small namespaces, so you can import only the JSONs that the page needs.

i18n.js
```js
module.exports = {
  pages: {
    '*': ['header', 'footer'],
    '/': ['home'],
    '/dashboard': ['dashboard'],
  },
}
```

header.json, footer.json, home.json & dashboard.json

**Reply by l-you** [reply]
Is there a tool to generate `pages` property automatically based on source code?

**Reply by aralroca** [reply]
No, but if you need it you probably can log the missing translations in each page using the log config field, and fetch to an API endpoint to add it in i18n.js during development
